shell.nix

```
@@ -1,159 +1,159 @@
-      alias c="./configure ${builtins.concatStringsSep " " configureFlags}"
-      alias c_no-wallet="./configure ${builtins.concatStringsSep " " configureFlags} --disable-wallet"
-      alias c_fast="./configure ${builtins.concatStringsSep " " configureFlags} --disable-wallet --disable-tests --disable-fuzz --disable-bench -disable-fuzz-binary"
-      alias c_fast_wallet="./configure ${builtins.concatStringsSep " " configureFlags} --disable-tests --disable-bench"
-
-      # make
-      alias m="make -j${jobs}"
-
-      # configure + make combos
-      alias cm="c && m"
-      alias cm_fast="c_fast && m"
-
-      # autogen + configure + make combos
-      alias acm="a && c && m"
-      alias acm_nw="a && c_no-wallet && m"
-      alias acm_fast="a && c_fast && m"
-      alias acm_fast_wallet="a && c_fast_wallet && m"
-
-      # tests
-      alias ut="make check"
-      # functional tests
-      alias ft="python3 test/functional/test_runner.py"
-      # all tests
-      alias t="ut && ft"
-
-      echo "adding ${builtins.concatStringsSep ":" binDirs} to \$PATH to make running built binaries more natural"
-      export PATH=$PATH:${builtins.concatStringsSep ":" binDirs};
-
-      alias a c m c_fast cm acm acm_nw acm_fast ut ft t
+  binDirs = [ "$PWD/build/src" "$PWD/build/src/qt" ];
+  jobs = if (strings.hasSuffix "linux" builtins.currentSystem) then
+    "$(($(nproc)-${toString spareCores}))"
+  else if (strings.hasSuffix "darwin" builtins.currentSystem) then
+    "$(($(sysctl -n hw.physicalcpu)-${toString spareCores}))"
+  else
+    "6";
+
+  libmultiprocess = pkgs.stdenv.mkDerivation {
+    name = "libmultiprocess";
+    src = pkgs.fetchFromGitHub {
+      owner = "bitcoin-core";
+      repo = "libmultiprocess";
+      rev = "f35df6bdc536b068597559d4ab470dab9cff7cfc";
+      sha256 = "sha256-1gg6MAql70JUXOdaP0A9Lmny5y6EFw55bJrzUvYRzMQ=";
+    };
+
+    nativeBuildInputs = [ pkgs.cmake pkgs.pkg-config ];
+    buildInputs = [ pkgs.capnproto ];
+    cmakeFlags =
+      [ "-DCMAKE_INSTALL_LIBDIR=lib" "-DCMAKE_INSTALL_INCLUDEDIR=include" ];
+
+    # Optional test check
+    doCheck = true;
+    checkPhase = ''
+      make check
+    '';
+
+    # Make sure pkg-config and cmake files are properly installed
+    postInstall = ''
+      mkdir -p $out/lib/pkgconfig
     '';
+
+    # Makes sure that the pkg-config and cmake files can find dependencies
+    setupHook = pkgs.writeText "setup-hook.sh" ''
+      export PKG_CONFIG_PATH="''${PKG_CONFIG_PATH:+$PKG_CONFIG_PATH:}${
+        placeholder "out"
+      }/lib/pkgconfig"
+      export CMAKE_PREFIX_PATH="''${CMAKE_PREFIX_PATH:+$CMAKE_PREFIX_PATH:}${
+        placeholder "out"
+      }"
+    '';
+
+    meta = with pkgs.lib; {
+      description = "Multi-process library";
+      homepage = "https://github.com/bitcoin-core/libmultiprocess";
+      license = licenses.mit;
+    };
+  };
+in pkgs.mkShell {
+  nativeBuildInputs = with pkgs; [
+    # Essential build tools
+    boost
+    ccache
+    clang-tools_19
```

CI complains about this.. maybe
```suggestion
    llvmPackages_19.clang-tools
```

or just 
```suggestion
    clang-tools
```